Documentation/upgrade.md

```diff
@@ -0,0 +1,293 @@
+rook-ceph-mon2   10.3.0.197:6790/0   true        OK
+
+MGRs:
+NAME             STATUS
+rook-ceph-mgr0   Active
+
+OSDs:
+TOTAL     UP        IN        FULL      NEAR FULL
+6         6         6         false     false
+
+PLACEMENT GROUPS (100 total):
+STATE          COUNT
+active+clean   100
+```
+In the output above, note the following indications that the cluster is in a healthy state:
+* Overall status: The overall cluster status is `OK` and there are no warning or error status messages displayed.
+* Monitors:  All of the monitors are `in quorum` and have individual status of `OK`.
+* OSDs: All OSDs are `UP` and `IN`.
+* MGRs: All Ceph managers are in the `Active` state.
+* Placement groups: All PGs are in the `active+clean` state.
+
+If your `rookctl status` output has deviations from the general good health described above, there may be an issue that needs to be investigated further.
+
+### Pod Version
+The version of a specific pod in the Rook cluster can be verified in its pod spec output.  For example, for the monitor pod `mon0`, we can verify that it is running `v0.5.1` with the below commands:
+```bash
+MON0_POD_NAME=$(kubectl -n rook get pod -l mon=rook-ceph-mon0 -o jsonpath='{.items[0].metadata.name}')
+kubectl -n rook get pod ${MON0_POD_NAME} -o jsonpath='{.spec.containers[0].image}'
+```
+
+### All Pods Status and Version
+The status and version of all Rook pods can be collected all at once with the following commands:
+```bash
+kubectl get pod -l app=rook-operator -o jsonpath='{range .items[*]}{.metadata.name}{"\n\t"}{.status.phase}{"\n\t"}{.spec.containers[0].image}{"\n"}{end}'
+kubectl -n rook get pod -o jsonpath='{range .items[*]}{.metadata.name}{"\n\t"}{.status.phase}{"\n\t"}{.spec.containers[0].image}{"\n"}{end}'
+```
+
+### Rook Volume Health
+Any pod that is using a Rook volume should also remain healthy:
+* The pod should be in the `Running` state with no restarts
+* There shouldn't be any errors in its logs
+* The pod should still be able to read and write to the attached Rook volume.
+
+## Upgrade Process
+The general flow of the upgrade process will be to upgrade the version of a Rook pod, verify the pod is running with the new version, then verify that the overall cluster health is still in a good state.
+
+In this guide, we will be upgrading a live Rook cluster running `v0.5.0` to the next available version of `v0.5.1`.
+Let's get started!
+
+### Operator
+The Rook operator is the management brains of the cluster, so it should be upgraded first before other components.
+In the event that the new version requires a migration of metadata or config, the operator is the one that would understand how to perform that migration.
+
+The operator is managed by a Kubernetes deployment, so in order to upgrade the version of the operator pod, we will need to edit the image version of the pod template in the deployment spec.  Begin editing the deployment with:
+```bash
+kubectl edit deployment rook-operator
```

Comment:
One simpler way to do this is running `$ kubectl set image deployment/rook-operator rook-operator=rook/rook:1.9.1`

Reply:
That definitely simplifies the instructions, I didn't know about that command.  I will definitely incorporate it and it will make the guide flow more easily and be more simple overall.  Thanks!